vars/withSubscriptionSp.groovy

```diff
@@ -0,0 +1,74 @@
+#!groovy
+import groovy.json.JsonSlurperClassic
+
+def call(String subscription, Closure body) {
```

Comment:
this doesn't appear to be called? can you explain this? possibly detach it to a separate PR as this is the only blocker from me right now

Reply:
![image](https://user-images.githubusercontent.com/47384516/118081350-17845b00-b3b3-11eb-9ac1-87a77b8e346f.png)

[Jenkinsfile_CNP](https://github.com/hmcts/cnp-idam-packer/blob/d43797286cdd8989d08c0f7b731e4de7ff6acb23/Jenkinsfile_CNP)
[Jenkinsfile_parameterized](https://github.com/hmcts/cnp-idam-packer/blob/d43797286cdd8989d08c0f7b731e4de7ff6acb23/Jenkinsfile_parameterized)
[README.md](https://github.com/hmcts/cnp-idam-master/blob/cbd38b355f53620d61bba24de64abef70da769a6/README.md)
It is being used, but .. no I can't explain it, Henry Dobson and Douglas Brand Williamson implemented it. 
It just seems like they need a specific set of variables set to do.. something..

Reply:
if I compare it with the standard **_withSubscription_** the difference is this:
![image](https://user-images.githubusercontent.com/47384516/118082156-962dc800-b3b4-11eb-839f-e2f91eca4a3c.png)
 you always call 
```groovy
        servicePrincipalBasedLogin(subscription, body)
```
its otherwise identical

Reply:
I think that was because packer didn't support managed identity, which looks like it might right now, https://www.packer.io/docs/builders/azure#azure-managed-identity

Again can we detach to this to a different PR? As I think this shouldn't be needed now, support for this was intentionally removed.

Reply:
sure, I will try re-introduce it if it turns out we cannot manage without it